python/lsst/ts/salobj/topics/read_topic.py

```diff
@@ -476,15 +482,22 @@ def flush(self):
             raise RuntimeError("Not allowed because there is a callback function")
         self._data_queue.clear()
 
-    def get(self, flush=True):
-        """Get the most recently seen message, or `None` if no data ever seen.
+    def get(self, flush=None):
+        """Get the most recent message, or `None` if no data available
+        (`has_data` False).
+
+        This method does not read from the queue, though it optionally
+        flushes the queue.
```

Comment:
Maybe add that this functionality will be deprecated in the future?

Reply:
What is deprecated is calling `get` without specifying the `flush` argument.

I could be convinced to deprecate allowing `get` to flush the queue at all. But the code and documentation would have to be written differently to do that. Being able to flush is occasionally convenient because it's a "best effort" and is allowed (but ignored) if there is a callback function, whereas the `flush` method raises an exception if there is a callback. Still...users should know whether there is a callback or not so it's not that big a deal.

Reply:
I thought about this some more and decided to deprecate the capability for get to flush the queue. Unfortunately this has to be done in stages (to avoid breaking code), and the first stage is to issue a deprecation warning if flush=True, or if the flush argument is omitted, since True is the default. I implemented that here.

If we want to be fully careful then there are 3 more stages (for later versions of salobj):
* Raise if flush=True (or the argument is omitted). We can skip this step if we can prove that all code has been updated to use flush=False.
* Make the default False and warn if the argument is explicitly specified. Raise if the flush=True.
* Eliminate the argument.